### PR TITLE
chore(all): merge master into V3

### PR DESCRIPTION
**Pull request by platypii**
Merge `master` into `V3` branch. These are nice changes, we should include them on the V3 branch.

This includes changes from [ee91f146 to b7e31802](https://github.com/jscad/OpenJSCAD.org/compare/ee91f146...b7e31802) which are all the commits to `master` since `V3` was forked from commit ee91f146.

Changes include:
 - cli options for multi part models
 - x3d-serializer fixes
 - performance and usability tweaks to modeling
 - svg-deserializer fix
 - more bezier functions
 - extrudeHelical

In the merge I had to port a bunch of these to ES modules, and fix the tests.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [x] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
@platypii cool! This will bring V3 up to the latest changes, and g provide a nice starting point for 'alpha' versions.

it may take a few days to complete the reviews.

File: packages/modeling/src/measurements/index.d.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
-export { measureAggregateArea } from './measureAggregateArea'
-export { measureAggregateBoundingBox } from './measureAggregateBoundingBox'
-export { measureAggregateEpsilon } from './measureAggregateEpsilon'
-export { measureAggregateVolume } from './measureAggregateVolume'
-export { measureArea } from './measureArea'
-export { measureBoundingBox } from './measureBoundingBox'
-export { measureEpsilon } from './measureEpsilon'
-export { measureVolume } from './measureVolume'
+export { default as measureAggregateArea } from './measureAggregateArea'
```
Comment (z3dev) [inline packages/modeling/src/measurements/index.d.ts:1]: I thought that we decided to only use named exports. Let's eliminate default

File: packages/modeling/src/operations/extrusions/extrudeHelical.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import test from 'ava'
+import { TAU } from '../../maths/constants.js'
+import { geom2, geom3 } from '../../geometries/index.js'
+import { circle } from '../../primitives/circle.js'
+
+import { extrudeHelical } from './extrudeHelical.js'
```
Comment (z3dev) [inline packages/modeling/src/operations/extrusions/extrudeHelical.test.js:6]: For tests... imports should be from index, as that verifies that the index is functionally correct.

File: packages/modeling/src/operations/hulls/hullPoints2.js [hunk 3]
```diff
@@ -25,11 +25,9 @@ export const hullPoints2 = (uniquePoints) => {
   })
 
   // sort by polar coordinates
-  points.sort((pt1, pt2) => pt1.angle < pt2.angle
-    ? -1
-    : pt1.angle > pt2.angle
-      ? 1
-      : pt1.distSq < pt2.distSq ? -1 : pt1.distSq > pt2.distSq ? 1 : 0)
+  points.sort((pt1, pt2) => pt1.angle !== pt2.angle
```
Comment (z3dev) [inline packages/modeling/src/operations/hulls/hullPoints2.js:28]: I don't remember this change... do you?

**Review by z3dev — COMMENTED** [review]
@platypii super job. I'm surprised that the changes merged so well. Bravo

but I have a few comments. Also, can you back out the changes to the packages in the dist directories? It's just noise because V3 will build and publish new versions.

File: packages/modeling/src/operations/hulls/hullPoints2.js (hunk 3 above)
Comment (platypii) [reply packages/modeling/src/operations/hulls/hullPoints2.js:28]: Yea that was me in the linting commit e1e20d1

**Comment by platypii** [comment]
Reverted the changes in `dist`, and addressed your other comments. Good catches thanks.

**Review by z3dev — APPROVED** [review]
@platypii thanks for the nice work again.

**Comment by platypii** [comment]
In the future, when its a branch merge like this, we should merge the commits in **without** "squash and merge". By squashing, git doesn't realize that master and V3 got synchronized, and lost track of the merge history :disappointed: 

V3-merge2

**Pull request merged**